Remotion/Web/Core/UI/SmartPage.cs

```diff
@@ -525,6 +525,11 @@ protected override void OnInit (EventArgs e)
   {
     base.OnInit(e);
     RegisterRequiresControlState(this);
+
```

Comment:
This implementation belongs in SmartPageInfo.  SmartPage doesn't actually have any logic itself. Originally, the idea was that anyone can implement the ISmartPage interface on a custom base-Page by delegating to the SmartPageInfo type.